src/domain/app/App.tsx

```diff
@@ -26,14 +28,17 @@ const App: React.FC = () => {
           <Redirect exact from="/" to="/harbors" />
           <Page>
             <Switch>
-              <PrivateRoute exact path="/harbors/:id" component={IndividualHarborPage} />
-              <PrivateRoute exact path="/harbors" component={HarborsPage} />
-              <PrivateRoute exact path="/customers/:id" component={IndividualCustomerPage} />
-              <PrivateRoute exact path="/customers" component={CustomersPage} />
-              <PrivateRoute exact path="/applications/:id" component={IndividualApplicationPage} />
-              <PrivateRoute exact path="/applications" component={ApplicationsPage} />
-              <PrivateRoute exact path="/offer/:applicationId" component={OfferPage} />
-              <PrivateRoute exact path="/pricing" component={PricingPage} />
+              <Route exact path="/error" component={ErrorPage} />
+              <ErrorBoundary errorComponent={<Redirect to="/error" />}>
+                <PrivateRoute exact path="/harbors/:id" component={IndividualHarborPage} />
+                <PrivateRoute exact path="/harbors" component={HarborsPage} />
+                <PrivateRoute exact path="/customers/:id" component={IndividualCustomerPage} />
+                <PrivateRoute exact path="/customers" component={CustomersPage} />
+                <PrivateRoute exact path="/applications/:id" component={IndividualApplicationPage} />
+                <PrivateRoute exact path="/applications" component={ApplicationsPage} />
+                <PrivateRoute exact path="/offer/:applicationId" component={OfferPage} />
+                <PrivateRoute exact path="/pricing" component={PricingPage} />
```

Comment:
These `PrivateRoute`s should be wrapped directly by a `Switch` component; otherwise, the router would try to render all of them.
You can test it, for example, by adding a `console.log('private route')` statement inside the `PrivateRoute` component, it will be called eight times in this case, while if we wrap them with a `Switch` component, it will be called only once.

Reply:
Fixed.